### DISCUSSION TITLE
HTTP status 403 after commiting code into repository

### DISCUSSION THREAD

**Discussion by roodee** [community/community#66308] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Dear community,

I am not sure if this is the right discussion category. 

My topic: I am facing for the first time issues when committing my code. This is how everything went:
•	Created a repository, governed by our organization
•	Created branch therein
•	Created GitHub codespace for this branch
•	Created code
•	Ran `git add …`
•	Running `git commit -m …` results in error message:

```
error: gpg failed to sign the data:
[GNUPG:] BEGIN_SIGNING
2023/09/07 07:39:05 error signing commit: error signing commit: error making request: 403 | Author is invalid, error making request: 403 | Author is invalid

fatal: failed to write commit object
```

Further information, which might help to find the solution

OS
```
$ uname -a
Linux codespaces-41950c 5.15.0-1041-azure #48-Ubuntu SMP Tue Jun 20 20:34:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

Git version
```
$ git --version

git version 2.41.0
```

Git configuration
```
$ git config --list

filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
credential.helper=/.codespaces/bin/gitcredential_github.sh
user.name=...
user.email=...
gpg.program=/.codespaces/bin/gh-gpgsign
init.defaultbranch=main
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=https://github.com/...
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.edit_readme.remote=origin
branch.edit_readme.merge=refs/heads/edit_readme
commit.gpgsign=true
lfs.repositoryformatversion=0
```

This is the very first time I met this error message. I already tried creating a new codespace instance in a new branch and running the same procedure: Same effect.

What could be the reason? Thanks in advance.

Kind regards

roodee

**Comment by laittis** [comment]
I've had some timeouts today too, might be issues on github side related to earlier incidents this week?

**Reply by roodee** [reply]
Thanks for your answer. As written above: I am having the issue for the whole week already. So it's not just today.

**Comment by roodee** [comment, marked as answer]
Thanks to all. For what reason ever I may commit to my repo again. I will close the question.

**Reply by AlexDev404** [reply]
The error means that you don't have your `git user.name` set to the name you put on your profile.
You can check what you have it set to in [settings](https://github.com/settings/profile)

![image](https://github.com/user-attachments/assets/a2bf60bc-8e58-4378-8445-2b90a065961e)

You can fix it by running `git config --global user.name [NAME_IN_SETTINGS]`

Afterwards, the error should go away.

**Comment by Hypergang02** [comment]
How to fix this issue? I can't
